libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java

```diff
@@ -732,6 +733,26 @@ public synchronized void setCustomDomainInferencePattern(@Nullable Pattern patte
         this.customDomainInferencePattern = pattern;
     }
 
+    /**
+     * Return the custom tab browser to use during advanced authentication
+     * NB: the default browser on the device is used is null is specified or if the specified browser
```

Comment:
Should be `if` instead of `is` in "the default browser on the device is used **_if_** null".

Reply:
What happens if null is set and the default browser does not support custom tabs?

Reply:
What happens today if you don't have chrome installed and the default browser does not support custom tab?

Reply:
If you have Adv Auth properly setup with the intent filter in the app manifest it does not use the default browser.  It forces Chrome or fails with a toast message telling you to install chrome.

Reply:
Now with the code change, it will only fail and show a toast if the requested browser is not found.

If you did not request a specific browser (by setting customTabBrowser to null), no error is thrown because doesBrowserExist() returns false (see [here](https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2481/files#diff-33ae398946ac646b4e530934b50bfef16db75f0d5b0ce6be67d86c29cae155fcR418)) and no package is set on the intent, causing the default browser to launch and load the login url.

If the default browser does not support custom tab, the flow will fail. Do we have an example of browser that doesn't support custom tab?

Reply:
I think all of the major browsers support Custom Tabs so maybe it's not a big deal.  However, I think it would be a little better to get the default browsers package name and [check if it is in the list of apps that support Custom Tabs installed on the device](https://developer.chrome.com/docs/android/custom-tabs/howto-custom-tab-check/).